packages/ifpack2/src/Ifpack2_LocalSparseTriangularSolver_def.hpp

```diff
@@ -358,8 +359,75 @@ initialize ()
     (! G->isFillComplete (), std::runtime_error, "If you call this method, "
      "the matrix's graph must be fill complete.  It is not.");
 
+  if (A_crs_->isUpperTriangular() && htsImpl_.is_null()) {
+    // Reverse the storage for an upper triangular matrix
+    using local_matrix_type = typename crs_matrix_type::local_matrix_type;
+
+    auto Alocal = A_crs_->getLocalMatrix();
+    auto ptr    = Alocal.graph.row_map;
+    auto ind    = Alocal.graph.entries;
+    auto val    = Alocal.values;
+
+    auto numRows = Alocal.numRows(), numCols = Alocal.numCols(), numNnz = Alocal.nnz();
+
+    typename decltype(ptr)::non_const_type  newptr ("ptr", ptr.dimension_0 ());
+    typename decltype(ind)::non_const_type  newind ("ind", ind.dimension_0 ());
+    decltype(val)                           newval ("val", val.dimension_0 ());
```

Comment:
This should work for now, because we assume UVM.  However, please do insert an `execution_space::fence()` before and after the setup phase, just to make sure that the GPU sees the changes made on host.  Also, please add a comment saying something like "FIXME this assumes UVM."  Thanks!